keepassxc-browser/content/banner.js

```diff
@@ -141,11 +136,12 @@ kpxcBanner.create = async function(credentials = {}) {
 };
 
 kpxcBanner.saveNewCredentials = async function(credentials = {}) {
-    const result = await browser.runtime.sendMessage({
-        action: 'get_database_groups'
-    });
+    const result = await sendMessage('get_database_groups');
 
-    if (!result.defaultGroupAlwaysAsk && (result.defaultGroup !== '' && result.defaultGroup !== DEFAULT_BROWSER_GROUP)) {
+    if (!result.defaultGroupAlwaysAsk
+        && result.groups
+        && result.groups.length > 0
+        && (result.defaultGroup !== '' && result.defaultGroup !== DEFAULT_BROWSER_GROUP)) {
```

Comment:
I find these if statements to be incomprehensible.

Reply:
Me too. And I'm still trying to simplify them somehow.